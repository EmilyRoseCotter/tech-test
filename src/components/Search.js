import React, { Fragment, useState } from "react";
import "../styles/Search.css";

function Search() {
  const [value, setValue] = useState();

  return(
      <>
      <form className="search-form"> 
      <input className="search_input" type="text" onChange={(e) => setValue(e.target.value)}/>
      <button type="submit" className="search-button" >Go!</button>
      </form>
      </>
  )
}

export default Search;