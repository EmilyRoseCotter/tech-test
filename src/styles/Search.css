.search-form {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.search-input {
  width: 20%;
  height: 30px;
  border-radius: 11px;
  font-size: 15px;
  padding: 0 10px;
}

.search-button {
  border-radius: 11px;
  width: 50px;
  color: #fff;
  background-color: #D0312D;
  margin-left: 5px;
}