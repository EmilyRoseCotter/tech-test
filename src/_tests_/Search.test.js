import React from "react";
import { render, screen } from '@testing-library/react';
import Search from '../components/Search.js';

describe("Search", () => {
  const validProps = {
    setSearchResults: () => {}
  };

  it("renders Search component correctly", () => {
    const { asFragment } = render(
      <Search 
      results={validProps.setSearchResults}
      />)
      ;

    expect(asFragment()).toMatchSnapshot(); 
  });

  it("button displays the correct text", () => {
    render(
      <Search 
      results={validProps.setSearchResults}
      />
    );
    const buttonElement = screen.getByText("Go!")

    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveClass("search-button");
  });
});