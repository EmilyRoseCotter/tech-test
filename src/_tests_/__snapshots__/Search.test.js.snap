// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search redners Search component correctly 1`] = `
<DocumentFragment>
  <div
    class="Search"
  >
    <form
      class="search-form"
    >
      <input
        class="search-input"
        type="text"
      />
      <button
        class="search-button"
        type="submit"
      >
        Go!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
